notebooks/official/workbench/spark/spark_sample_notebook.ipynb

```diff
@@ -0,0 +1,1200 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "WReHDGG5g0XY"
+   },
+   "source": [
+    "### Set your project ID\n",
+    "\n",
+    "Run the following cell to try to get you project ID."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "id": "oM1iC_MfAts1"
+   },
+   "outputs": [],
+   "source": [
+    "import os\n",
+    "\n",
+    "PROJECT_ID = \"\"\n",
+    "\n",
+    "# Try to get your Google Cloud project ID from gcloud\n",
+    "if not os.getenv(\"IS_TESTING\"):\n",
+    "    shell_output = !gcloud config list --format 'value(core.project)' 2>/dev/null\n",
+    "    PROJECT_ID = shell_output[0]\n",
+    "    print(\"Project ID: \", PROJECT_ID)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "qJYoRfYng0XZ"
+   },
+   "source": [
+    "If the previous command has no output, copy your project ID using the project selector in the [Google Cloud console](https://console.cloud.google.com/). Insert the ID in the `[your-project-id]` placeholder, then run the following command."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "id": "riG_qUokg0XZ"
+   },
+   "outputs": [],
+   "source": [
+    "if PROJECT_ID == \"\" or PROJECT_ID is None:\n",
+    "    PROJECT_ID = \"[your-project-id]\"  # @param {type: \"string\"}"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "id": "set_gcloud_project_id"
+   },
+   "outputs": [],
+   "source": [
+    "! gcloud config set project $PROJECT_ID -q"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "zgPO1eR3CYjk"
+   },
+   "source": [
+    "### Create a Cloud Storage bucket\n",
+    "\n",
+    "The Spark DataFrame created in this tutorial is stored in BigQuery, with the data first being written to a Google Cloud Storage bucket before it is written into BigQuery."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "region"
+   },
+   "source": [
+    "#### Region\n",
+    "\n",
+    "In order to create a bucket, you must define the `REGION` variable. Choose [the region closest to you](https://cloud.google.com/compute/docs/regions-zones)."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "id": "region"
+   },
+   "outputs": [],
+   "source": [
+    "REGION = \"[your-region]\"  # @param {type: \"string\"}\n",
+    "\n",
+    "if REGION == \"[your-region]\":\n",
+    "    REGION = \"us-central1\""
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "06571eb4063b"
+   },
+   "source": [
+    "#### Timestamp\n",
+    "\n",
+    "To avoid name collisions, you create a timestamp for each notebook session, then append it to the name of resources you create in this tutorial, which include a Cloud Storage bucket and BigQuery dataset."
```

Comment:
To avoid name collisions, you create a timestamp for each notebook session, then append it to the name of resources you create in this tutorial, which include a Cloud Storage bucket and BigQuery dataset. -> To avoid name collisions, you can create a timestamp for the current notebook session, then append the timestamp to the name of resources that you create in this tutorial, such as the Cloud Storage bucket or BigQuery dataset that you create in this tutorial.